lib/device.c

```diff
@@ -89,32 +89,36 @@ enum {
 	SAVE_TRACKS = 5
 };
 
-static inline int socket_poll(SOCKET socket)
+static inline int socket_poll(SOCKET socket, int out)
 {
 #ifdef GEKKO
 	// libogc doesn't impmelent select()...
 	struct pollsd sds[1];
 	sds[0].socket  = socket;
-	sds[0].events  = POLLIN;
+	sds[0].events  = out ? POLLOUT : POLLIN;
 	sds[0].revents = 0;
 	if (net_poll(sds, 1, 0) < 0) return 0;
-	return (sds[0].revents & POLLIN) && !(sds[0].revents & (POLLERR|POLLHUP|POLLNVAL));
+	return (sds[0].revents & (out ? POLLOUT : POLLIN)) && !(sds[0].revents & (POLLERR|POLLHUP|POLLNVAL));
```

Comment:
We're repeating ourselves a bit here. Maybe do something like `int events = out ? POLLOUT : POLLIN;` first instead?

Reply:
Yeah that should be cleaned up.

I was also considering always providing xfds to select() instead of only when !!out.  But decided against changing existing behavior like that, it's just the non-blocking connects that *need* the xfds on windows in particular.